app/uk/gov/hmrc/apisubscriptionfields/service/FieldsDefinitionService.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2017 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apisubscriptionfields.service
+
+import javax.inject.{Inject, Singleton}
+
+import play.api.Logger
+import uk.gov.hmrc.apisubscriptionfields.model._
+import uk.gov.hmrc.apisubscriptionfields.repository.{FieldsDefinition, FieldsDefinitionRepository}
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+
+@Singleton
+class FieldsDefinitionService @Inject() (repository: FieldsDefinitionRepository) {
+
+  def upsert(identifier: FieldsDefinitionIdentifier, fields: Seq[FieldDefinition]): Future[Boolean] = {
```

Comment:
One enhancement would be to change the return type as `Future[(SubscriptionFieldDefinitionResponse, Boolean)]`

See how `upsert`, `save`, `create` and `update` are implemented in `SubscriptionFieldsService`.

Reply:
The two services are different because the fields definition responses do not need to return a payload as part of their response. The subscription field endpoints **do** return a payload. 
Can we leave as-is?